optuna/visualization/_contour.py

```diff
@@ -242,13 +266,19 @@ def _generate_contour_subplot(
         y_values.append(y_value)
         x_i = x_indices.index(x_value)
         y_i = y_indices.index(y_value)
-        if isinstance(trial.value, int):
-            value = float(trial.value)
-        elif isinstance(trial.value, float):
+
+        if target is None:
             value = trial.value
         else:
+            value = target(trial)
+
+        if isinstance(value, int):
+            value = float(value)
+        elif not isinstance(value, float):
             raise ValueError(
-                "Trial{} has COMPLETE state, but its value is non-numeric.".format(trial.number)
+                "Trial{} has COMPLETE state, but its target value is non-numeric.".format(
+                    trial.number
+                )
```

Comment:
This is not directly related to this PR change, but what about using f-strings?
```suggestion
                f"Trial{trial.number} has COMPLETE state, but its target value is non-numeric."
```

Reply:
I also think it's better to use f-strings. Is it recommended in Optuna? Coding Style Conventions does not mention it.

Reply:
According to the issue https://github.com/optuna/optuna/issues/1916#issuecomment-708158413, we'd like to use f-strings when the line doesn't get too long. This is not a strict rule, but now, we can use f-strings without a too long line.